teps/0089-spire-support.md

```diff
@@ -0,0 +1,262 @@
+---
+status: proposed
+title: SPIRE Support
+creation-date: '2021-10-04'
+last-updated: '2021-11-18'
+authors:
+- '@priyawadhwa'
+---
+
+# TEP-0089: SPIRE Support
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Background](#background)
+- [SPIRE Concepts](#spire-concepts)
+  - [Installing SPIRE](#installing-spire)
+- [Proposed Solution](#proposed-solution)
+  - [1. Tekton Chains has no way to verify that a TaskRun in the cluster was created by Tekton](#1-tekton-chains-has-no-way-to-verify-that-a-taskrun-in-the-cluster-was-created-by-tekton)
+  - [2. Tekton Chains has no way to verify that the TaskRun it received wasn't modified by anybody other than Tekton, during or after execution](#2-tekton-chains-has-no-way-to-verify-that-the-taskrun-it-received-wasnt-modified-by-anybody-other-than-tekton-during-or-after-execution)
+  - [3. Tekton Pipelines can't verify that the results it reads weren't modified](#3-tekton-pipelines-cant-verify-that-the-results-it-reads-werent-modified)
+- [Motivation](#motivation)
+  - [Requirements](#requirements)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Implementation Plan](#implementation-plan)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Enabling SPIRE on the Entrypointer Image](#enabling-spire-on-the-entrypointer-image)
+  - [Verification in Chains](#verification-in-chains)
+- [Test Plan](#test-plan)
+- [Alternatives](#alternatives)
+  - [Kubernertes Service Account Token Volume Projection](#kubernertes-service-account-token-volume-projection)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP covers integrating Tekton and Tekton Chains with [SPIFFE/SPIRE](https://spiffe.io/), which would provide a more secure supply chain for Tekton users. It would also guarantee [non-falsifiable provenance](https://slsa.dev/requirements#non-falsifiable), which is a requirement for [SLSA Level 3](https://slsa.dev/levels).
+With this integration, Tekton will be one step closer to SLSA Level 3 compliance.
+
+## Background
+
+Currently, Tekton Chains observes Tekton and waits for TaskRuns to complete.
+Once it sees that a TaskRun has completed it tries to sign any artifacts that were built and also generates provenance.
+
+There are a couple issues with this:
+1. Tekton Chains has no way to verify that a TaskRun in the cluster was created by Tekton
```

Comment:
small detail - i think we're okay with a scenario where tekton didn't create the taskrun itself, e.g. it could be some triggering system that created it (even something like github actions), but the important thing is that the execution is carried out by the tekton pipelines controller, and all updates to the TaskRun once execution starts (or we could go as far as after creation) are done by the controller and not by something else (particularly the status)

Reply:
that makes sense! i think the second point pretty much covers that (that execution should be done by the controller only) so i'll remove this one.